cmd/thermal-recorder/main.go

```diff
@@ -140,6 +148,9 @@ func handleConn(conn net.Conn, conf *Config, turret *TurretController) error {
 			log.Printf("%d frames for this connection", totalFrames)
 		}
 
+		if throttledRecorder != nil {
+			throttledRecorder.NewFrame()
```

Comment:
If this was part of the Recorder interface (it would be a no-op for CPTVRecorder) then there would be no need for this if. You also wouldn't need to keep a separate reference to throttledRecorder. It could just wrap the CPTVRecorder and replace it.